tests/test_router_bgp.rb

```diff
@@ -733,22 +725,42 @@ def test_set_get_reconnect_interval
 
   def test_get_reconnect_interval_default
     skip('Not supported on IOS XR') if platform == :ios_xr
-    asnum = 55
-    bgp = RouterBgp.new(asnum)
+    bgp = setup_default
     assert_equal(60, bgp.reconnect_interval,
                  "reconnect_interval should be set to default value of '60'")
     bgp.destroy
   end
 
-  def test_set_get_router_id
+  def test_route_distinguisher
+    bgp = setup_vrf
+    bgp.route_distinguisher = 'auto'
+    assert_equal('auto', bgp.route_distinguisher,
+                 "bgp route_distinguisher should be set to 'auto'")
+    bgp.route_distinguisher = '1:1'
+    assert_equal('1:1', bgp.route_distinguisher,
+                 "bgp route_distinguisher should be set to '1:1'")
+    bgp.route_distinguisher = bgp.default_route_distinguisher
+    assert_empty(bgp.route_distinguisher,
+                 'bgp route_distinguisher should *NOT* be configured')
+    bgp.destroy
+  end
+
+  def test_default_route_distinguisher
+    bgp = setup_vrf
+    assert_empty(bgp.default_route_distinguisher,
+                 'bgp route_distinguisher default value should be empty')
+    bgp.destroy
+  end
+
+  def test_router_id
     %w(test_default test_vrf).each do |t|
       if t == 'test_default'
         bgp = setup_default
       else
         bgp = setup_vrf
       end
-      bgp.router_id = '1.2.3.4'
-      assert_equal('1.2.3.4', bgp.router_id,
+      bgp.router_id = '7.8.9.11'
+      assert_equal('7.8.9.11', bgp.router_id,
                    "vrf #{@vrf}: bgp router_id should be set to '1.2.3.4'")
```

Comment:
You changed the rid everywhere but in the error message.

IMO you don't need to state it in the rid at all since the assert is going to display \<expected\>'7.8.9.11' if it fails; ie. this would tell you everything already:

```
"vrf #{@vrf}: bgp router_id invalid")
```